### PR TITLE
Support for Linux

### PR DESCRIPTION
**Pull request by keheliya**
Hi @sindresorhus,
Let me know if I need to fix the coding style or anything else in this PR to match with the existing implementation. Thanks.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,30 +3,54 @@ const path = require('path');
 const childProcess = require('child_process');
 const pify = require('pify');
 const bin = path.join(__dirname, 'main');
+const xprop = "xprop -id $(xprop -root 32x '\t$0' _NET_ACTIVE_WINDOW | cut -f 2)";
```
Comment (sindresorhus) [inline index.js]: use single-quotes

File: index.js [hunk 2]
```diff
@@ -3,30 +3,54 @@ const path = require('path');
...
+		return {
+			title: parts[0] || null,
+			id: Number(parts[1]),
+			app: parts[2],
+			pid: Number(parts[3])
+		};
+	} else if (process.platform === 'linux') {
+		const result = {};
+		stdout.trim().split('\n').forEach(row => {
+			if (row.indexOf('=') > -1) {
```
Comment (sindresorhus) [inline index.js]: Use `row.includes` instead

File: index.js [hunk 3]
```diff
@@ -3,30 +3,54 @@ const path = require('path');
...
+				row = row.split('=');
+				result[row[0].trim()] = row[1].trim();
+			} else if (row.indexOf(':') > -1) {
```
Comment (sindresorhus) [inline index.js]: Use `row.includes` instead

File: index.js [hunk 4]
```diff
@@ -3,30 +3,54 @@ const path = require('path');
...
-	};
+		return {
+			title: parts[0] || null,
+			id: Number(parts[1]),
+			app: parts[2],
+			pid: Number(parts[3])
+		};
+	} else if (process.platform === 'linux') {
+		const result = {};
+		stdout.trim().split('\n').forEach(row => {
```
Comment (sindresorhus) [inline index.js]: Use a `for..of` loop

**Comment by sindresorhus** [comment]
Nice!

Can you also update the package.json description?

File: index.js [hunk 5]
```diff
@@ -3,30 +3,54 @@ const path = require('path');
...
+		});
+
+		return {
+			title: JSON.parse(result['_NET_WM_NAME(UTF8_STRING)']) || null,
+			id: parseInt(result['WM_CLIENT_LEADER(WINDOW)'].split('#').pop(), 16),
+			app: JSON.parse(result['WM_CLASS(STRING)'].split(',').pop()),
+			pid: parseInt(result['_NET_WM_PID(CARDINAL)'], 10)
+		};
+	}
+	throw new Error('OS X and Linux only');
```
Comment (sindresorhus) [inline index.js]: Add an empty line above this.

**Comment by keheliya** [comment]
Hi @sindresorhus,
I hope it's all good now and ready to be merged.

**Comment by sindresorhus** [comment]
Looks good. Thank you :)

Added you to the repo since this is an [open open source](http://openopensource.org) project :)

**Comment by sindresorhus** [comment]
I added Travis testing and seems to fail https://travis-ci.org/sindresorhus/active-win/jobs/123671909 Any idea? It works locally on my Ubuntu VM.

**Comment by keheliya** [comment]
I guess it's because the Travis job runs on a headless server and there's no Active Window. I'm looking into handling that.

**Comment by sindresorhus** [comment]
@keheliya Yes, but I'm running it with a fake window: https://github.com/sindresorhus/active-win/blob/521c5b99f25007bd082df16b19f3b6e2c0784028/.travis.yml#L9-L10

**Comment by keheliya** [comment]
It needed a Window Manager to be installed, and a new window to be spawned. The build is passing now.

**Comment by sindresorhus** [comment]
Awesome. Thanks for looking into that. 0.2.0 published.

Any idea why Travis is not testing on OS X? Multi-OS is defined in .travis.yml.

**Comment by keheliya** [comment]
Hi @sindresorhus,
I didn't get that. OS X build looks good to me.
screenshot 2016-04-20 at 7 31 47 am

**Comment by sindresorhus** [comment]
Weird. I probably missed it. Sorry for the noise.

**Pull request closed without merging**